server/src/main/java/org/elasticsearch/search/aggregations/metrics/max/MaxAggregator.java

```diff
@@ -118,4 +153,48 @@ public InternalAggregation buildEmptyAggregation() {
     public void doClose() {
         Releasables.close(maxes);
     }
+
+    /**
+     * Returns the maximum value indexed in the <code>fieldName</code> field or <code>null</code>
+     * if the value cannot be inferred from the indexed {@link PointValues}.
+     */
+    public static Number findLeafMaxValue(LeafReader reader, String fieldName, Function<byte[], Number> converter) throws IOException {
+        final PointValues pointValues = reader.getPointValues(fieldName);
+        if (pointValues == null) {
+            return null;
+        }
+        final Bits liveDocs = reader.getLiveDocs();
+        if (liveDocs == null) {
+            return converter.apply(pointValues.getMaxPackedValue());
+        }
+        int numBytes = pointValues.getBytesPerDimension();
+        final byte[] maxValue = pointValues.getMaxPackedValue();
+        final Number[] result = new Number[1];
+        pointValues.intersect(new PointValues.IntersectVisitor() {
+            @Override
+            public void visit(int docID) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void visit(int docID, byte[] packedValue) {
+                if (liveDocs.get(docID)) {
+                    // we need to collect all values in this leaf (the sort is ascending) where
+                    // the last live doc is guaranteed to contain the max value for the segment.
+                    result[0] = converter.apply(packedValue);
+                }
+            }
+
+            @Override
+            public PointValues.Relation compare(byte[] minPackedValue, byte[] maxPackedValue) {
+                if (FutureArrays.compareUnsigned(maxValue, 0, numBytes, maxPackedValue, 0, numBytes) == 0) {
```

Comment:
Arrays#equals?

Reply:
It's a java9 method, at least the one that takes a start and end offset, and I am not sure we can use the java8 `Arrays#equals(byte[], byte[])` since the min and max can have a length greater than `numBytes` ?

Reply:
I switched to `FutureArrays#equals` which is equivalent